src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,564 @@
+        ///   The name of a hashing algorithm, see <see href="https://github.com/multiformats/multicodec/blob/master/table.csv"/>
+        ///   for IPFS defined names.
+        /// </param>
+        /// <returns>
+        ///   The hashing implementation associated with the <paramref name="name"/>.
+        ///   After using the hashing algorithm it should be disposed.
+        /// </returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   When <paramref name="name"/> is not registered.
+        /// </exception>
+        public static HashAlgorithm GetHashAlgorithm(string name = DefaultAlgorithmName)
+        {
+            try
+            {
+                return HashingAlgorithm.Names[name].Hasher();
+            }
+            catch (KeyNotFoundException)
+            {
+                throw new KeyNotFoundException($"Hash algorithm '{name}' is not registered.");
+            }
+        }
+
+        /// <summary>
+        ///   Gets the name of hashing algorithm name with the specified code.
+        /// </summary>
+        /// <param name="code">
+        ///   The code of a hashing algorithm, see <see href="https://github.com/multiformats/multicodec/blob/master/table.csv"/>
+        ///   for IPFS defined codes.
+        /// </param>
+        /// <returns>
+        ///   The name assigned to <paramref name="code"/>.
+        /// </returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   When <paramref name="code"/> is not registered.
+        /// </exception>
+        public static string GetHashAlgorithmName(int code)
+        {
+            try
+            {
+                return HashingAlgorithm.Codes[code].Name;
+            }
+            catch (KeyNotFoundException)
+            {
+                throw new KeyNotFoundException($"Hash algorithm with code '{code}' is not registered.");
+            }
+        }
+
+        /// <summary>
+        ///   Occurs when an unknown hashing algorithm number is parsed.
+        /// </summary>
+        public static EventHandler<UnknownHashingAlgorithmEventArgs> UnknownHashingAlgorithm;
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="MultiHash"/> class with the
+        ///   specified <see cref="HashingAlgorithm">Algorithm name</see> and <see cref="Digest"/> value.
+        /// </summary>
+        /// <param name="algorithmName">
+        ///   A valid IPFS hashing algorithm name, e.g. "sha2-256" or "sha2-512".
+        /// </param>
+        /// <param name="digest">
+        ///    The digest value as a byte array.
+        /// </param>
+        public MultiHash(string algorithmName, byte[] digest)
+        {
+            if (algorithmName == null)
+                throw new ArgumentNullException("algorithmName");
+            if (digest == null)
+                throw new ArgumentNullException("digest");
+
+            if (!HashingAlgorithm.Names.TryGetValue(algorithmName, out var a))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)